frontend/packages/console-dynamic-plugin-sdk/src/utils/fetch/console-fetch-utils.ts

```diff
@@ -27,20 +27,23 @@ export const getCSRFToken = () => {
  */
 export const getConsoleRequestHeaders = (): ConsoleRequestHeaders => {
   const store = storeHandler.getStore();
-  if (!store) return undefined;
-  const state = store.getState();
 
   const headers: ConsoleRequestHeaders = {
     'X-CSRFToken': getCSRFToken(),
   };
 
-  // Set impersonation headers
-  const { kind, name } = getImpersonate(state) || {};
-  if ((kind === 'User' || kind === 'Group') && name) {
-    // Even if we are impersonating a group, we still need to set Impersonate-User to something or k8s will complain
-    headers['Impersonate-User'] = name;
-    if (kind === 'Group') {
-      headers['Impersonate-Group'] = name;
+  // If store is available, add impersonation headers
```

Comment:
I'm not sure this fixes the issue. We are re arranging the logic but the outcome is the same. I tested locally and the CSRF token still not populated. Is it maybe related to the initialization of the store, so is actually available from a plugin consumer?

Reply:
@jgbernalp CSRF token is not in the redux store, it's parsed from the cookies. Were the impersonation headers also missing. @Leo6Leo Should we make sure the `csrf-token` cookie is present on the document as expected here: https://github.com/openshift/console/blob/b33ff3ee10e5dcf076bee83e5dcba1de40900c6e/frontend/packages/console-dynamic-plugin-sdk/src/utils/fetch/console-fetch-utils.ts#L15